test/support.ts

```diff
@@ -153,14 +153,15 @@ export async function prepareTransactionTest(sequelize: Sequelize) {
 }
 
 export function createSequelizeInstance(options: Options = {}) {
-  options.dialect = getTestDialect();
+  options.dialect = getTestDialect(); // only returns translated dialectName "mysql8"->"mysql"
 
-  const config = Config[options.dialect];
+  // pass process.env.DIALECT b/c it may contain version "postgres12" config can use
+  const config = Config(process.env.DIALECT);
 
   const sequelizeOptions = defaults(options, {
     host: options.host || config.host,
     logging: process.env.SEQ_LOG ? console.debug : false,
-    dialect: options.dialect,
+    dialect: getDialectName(options.dialect),
```

Comment:
why calling `getDialectName` here? `getTestDialect` is supposed to return a valid `Dialect` already

Reply:
Good question.  

I don't know if this was oversight or because I thought making it obvious that `options.dialect` is the dialect name.  Looking at `createSequelizeInstance` I'm curious why it takes in `options` that are then overwritten by the environment variable (i.e. `process.env.DIALECT` via _getTestDialect_).

Because `options` is a "reference", it's values are mutable in the caller.  The caller might not want the _dialect_ to change in its options it passed in -- it could iterate and pass in `mysql5` and then want to try (locally `options.dialect+'7'` to test `mysql57`, but with function changing the value it'd be `mysql7`). We could deep copy options, but that might be overkill.

I think maybe this is all that's needed.  Use supplied options, or get dialect from environment:

   ```javascript
   options.dialect = getDialectName(options?.dialect||'') || getTestDialect(),
   ```


Sidenote:
1. I think the _Test_ in _getTestDialect_ is a little confusing or might benefit from a change to  `getDialectNameFromEnvironment`, which would be a little less confusing. I'm punting on this for now as all tests call the function.
2. `getDialectName` is a parser that takes any _dialectal_ string argument and returns a dialect or empty string, which means `getDialectName` is a little clearer and shows intent in the code